### PR TITLE
PIM-7720: add variant axis basics

### PR DESCRIPTION
**Pull request by juliensnz**
**Description (for Contributor and Core Developer)**

This PR adds support for reference entities attributes as variant axis

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Todo
| Added legacy Behats               | Todo
| Added acceptance tests            | Todo
| Added integration tests           | Todo
| Changelog updated                 | Todo
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Akeneo/Pim/Structure/Component/Model/FamilyVariant.php [hunk 1]
```diff
@@ -287,6 +287,7 @@ public static function getAvailableAxesAttributeTypes(): array
             AttributeTypes::OPTION_SIMPLE_SELECT,
             AttributeTypes::BOOLEAN,
             AttributeTypes::REFERENCE_DATA_SIMPLE_SELECT,
+            'akeneo_reference_entity'
```
Comment (grena) [inline src/Akeneo/Pim/Structure/Component/Model/FamilyVariant.php]: No const?

File: src/Akeneo/Platform/Bundle/UIBundle/Resources/config/form_extensions/product_model/add_child.yml [hunk 2]
```diff
@@ -18,6 +18,7 @@ extensions:
                 akeneo-switch-field: pim-product-model-add-child-field-switch
                 akeneo-simple-select-reference-data-field: pim-product-model-add-child-field-simple-select-reference-data
                 akeneo-text-field: pim-product-model-add-child-field-text
+                akeneo-reference-entity-field: pim-product-model-add-child-field-simple-select-reference-entity
```
Comment (grena) [inline src/Akeneo/Platform/Bundle/UIBundle/Resources/config/form_extensions/product_model/add_child.yml]: `akeneo-simple-select-reference-entity-field:` to be consistent with `akeneo-simple-select-reference-data-field`

File: src/Akeneo/Platform/Bundle/UIBundle/Resources/config/form_extensions/product_model/add_child.yml [hunk 3]
```diff
@@ -39,6 +40,13 @@ extensions:
         module: pim/form/common/fields/values/simple-select-async
         config:
             required: true
+            route:
```
Comment (grena) [inline src/Akeneo/Platform/Bundle/UIBundle/Resources/config/form_extensions/product_model/add_child.yml]: Normal to have an empty route here?

File: src/Akeneo/Pim/Structure/Component/Model/FamilyVariant.php (hunk 1 above)
Comment (juliensnz) [reply src/Akeneo/Pim/Structure/Component/Model/FamilyVariant.php]: sorry, fixed it but not pushed

**Pull request merged**